OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/util/KeyFormattingUtils.java

```diff
@@ -271,6 +271,25 @@ private static String convertKeyIdToHex32bit(long keyId) {
         return hexString;
     }
 
+    /**
+     * Convert key id from hex string to long
+     * @param keyIdHex - 64 bit hex string representation of keyId, including "0x" prefix
+     * @return keyId as long
+     */
+    public static Long convertKeyIdHexToLong(String keyIdHex) {
+        if (keyIdHex == null) { return null;}
+
+        //we can't convert directly from hex
```

Comment:
It appears you can't parse a long from a two's complement encoded hex string using Long.parseLong or Long.decode http://stackoverflow.com/questions/1410168/how-to-parse-negative-long-in-hex-in-java

I can make the comment a bit more verbose if you want?